assets/src/edit-story/panels/panel/handle.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useRef,
+	useState,
+	useCallback,
+	useLayoutEffect,
+} from '@wordpress/element';
+
+const Handle = styled.button.attrs( { type: 'button' } )`
+	background: transparent;
+	border: 0;
+	padding: 0;
+	position: absolute;
+	top: 0;
+	left: 50%;
+	margin-left: -60px;
+	width: 120px;
+	height: 20px;
+	display: flex;
+	flex-direction: column;
+	justify-content: flex-start;
+	align-items: center;
+	cursor: row-resize;
+`;
+
+const Bar = styled.div`
+	margin-top: 4px;
+	background-color: ${ ( { theme } ) => theme.colors.bg.v0 };
+	width: 32px;
+	height: 4px;
+	border-radius: 2px;
+`;
+
+function DragHandle( { handleHeightChange } ) {
+	const handle = useRef();
+	const lastPosition = useRef();
+	const [ isDragging, setIsDragging ] = useState( false );
+
+	// On mouse move, check difference since last record vertical mouse position
+	// and invoke callback with this difference.
+	// Then record new vertical mouse position for next iteration.
+	const handleMouseMove = useCallback( ( evt ) => {
+		const delta = lastPosition.current - evt.pageY;
+		handleHeightChange( delta );
+		lastPosition.current = evt.pageY;
+	}, [ handleHeightChange ] );
+
+	// On mouse up, set dragging as false
+	// - will cause useLayoutEffect to unregister listeners.
+	const handleMouseUp = useCallback( () => setIsDragging( false ), [] );
+
+	// On mouse down, set dragging as true
+	// - will cause useLayoutEffect to register listeners.
+	// Also record the initial vertical mouse position on the page.
+	const handleMouseDown = useCallback( ( evt ) => {
+		lastPosition.current = evt.pageY;
+		setIsDragging( true );
+	}, [] );
+
+	// On initial render *and* every time `isDragging` changes value,
+	// register all relevant listeners. Note that all listeners registered
+	// will be correctly unregistered due to the cleanup function.
+	useLayoutEffect( () => {
+		const element = handle.current;
+		const doc = element.ownerDocument;
+		element.addEventListener( 'mousedown', handleMouseDown );
+
+		if ( isDragging && doc ) {
+			doc.addEventListener( 'mousemove', handleMouseMove );
+			doc.addEventListener( 'mouseup', handleMouseUp );
+		}
+
+		return () => {
+			if ( element ) {
+				element.removeEventListener( 'mousedown', handleMouseDown );
+				if ( isDragging && doc ) {
```

Comment:
Testing this right now and seeing this running hundreds of times while dragging. Just to confirm -- is this expected? E.g. with console.log:
<img width="1395" alt="Screenshot 2020-01-20 at 20 22 03" src="https://user-images.githubusercontent.com/3294597/72752461-8b00c880-3bc2-11ea-8f28-e7b7b24895ee.png">

Reply:
Definitely not on purpose. Haven't checked for optimizations yet, and there's clearly something being redefined on every render.

Reply:
Fixed in 9a65c698c